core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1470,6 +1470,7 @@ private Scope createScopeForMaterializedView(Table table, QualifiedObjectName na
 
         private Scope createScopeForView(Table table, QualifiedObjectName name, Optional<Scope> scope, ConnectorViewDefinition view)
         {
+            checkArgument(view.isRunAsInvoker() || view.getOwner().isPresent(), "owner must be defined with a view with SECURITY DEFINER mode");
```

Comment:
Let's change this to throw a better error, since it means the connector returned a bad view definition
```java
if (!view.isRunAsInvoker() && view.getOwner().isEmpty()) {
    throw semanticException(INVALID_VIEW, table, "Owner must be present in view with SECURITY DEFINER mode: %s", name);
}